lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -272,7 +272,8 @@ def range_for_autocorrect_ruby19(pair_node)
         end
 
         def argument_without_space?(node)
-          node.argument? && node.source_range.begin_pos == node.parent.loc.selector.end_pos
+          node.argument? && node.parent.loc.selector &&
+            node.source_range.begin_pos == node.parent.loc.selector.end_pos
```

Comment:
Can you tweak it to the following?

```suggestion
          return false if !node.argument? || !node.parent.loc.selector

          node.source_range.begin_pos == node.parent.loc.selector.end_pos
```